File: src/com/anathema_roguelike/stats/Stat.java
package com.anathema_roguelike.stats;

public abstract class Stat<T> {
	
	private T object;
	
	public Stat(T object) {
		this.object = object;
	}
	
	public T getObject() {
		return object;
	}
	
	public abstract double getAmount();
}

File: src/com/anathema_roguelike/stats/Stat.scala


package com.anathema_roguelike
package stats

import com.anathema_roguelike.entities.characters.Character
import com.anathema_roguelike.environment.Location

object Stat {
  type CharacterStat = Stat[Character]
  type LocationStat = Stat[Location]
}

abstract class Stat[T](var obj: T) {
  def getObject: T = obj

  def getAmount: Double
}
